totalRP3/core/impl/slash.lua

```diff
@@ -167,6 +240,48 @@ end
 TRP3_API.events.listenToEvent(TRP3_API.events.WORKFLOW_ON_LOADED, function()
 
 
+	TRP3_API.slash.registerConditional({
+		id = "ic",
+		test = function()
+			return (TRP3_API.profile.getData("player/character/RP") == 1);
+		end
+	})
+	TRP3_API.slash.registerConditional({
+		id = "ooc",
+		test = function()
+			return (TRP3_API.profile.getData("player/character/RP") ~= 1);
+		end
+	})
+
+	local location = function(id, targetLocation)
+		--we don't allow spaces in conditionals. So we turn any multiword names into camelCase
+		local locationString = GetMinimapZoneText();
+		locationString = string.lower(locationString);--we lowercase it all
+		locationString = string.gsub(locationString, "[^%s%a]", "");--remove any apostrophies (or anything else that's not a letter)
+		locationString = string.gsub(locationString,"(%s)(%a)", function(a,b) return string.upper(b) end) -- and turn it into a camelCaseValue
+		return targetLocation==locationString; --then compare it to the given string
+	end
+	TRP3_API.slash.registerConditional({
+		id = "loc",
+		test = location
+	})
+	TRP3_API.slash.registerConditional({

```

Comment:
I'd favour having one condition rather than two for location, ignoring the fact that `[stance]`, `[form]`, etc. are a mess.

Personally I'd favour `[location:goldshire]` over `loc` for a condition. Stands out more when you read it.

Reply:
Location is the prefered form. However, thanks to the character limit in conditionals, it's preferable to have a short form of long conditions. I.e. location can be shortened to loc. Blizzard works under the same theory with things like modifier -> mod